nomad/structs/structs.go

```diff
@@ -1797,6 +1806,29 @@ func (r *Resources) Add(delta *Resources) error {
 	return nil
 }
 
+// Subtract removes the resources of the delta to this, potentially
+// returning an error if not possible.
+func (r *Resources) Subtract(delta *Resources) error {
+	if delta == nil {
+		return nil
+	}
+	r.CPU -= delta.CPU
+	r.MemoryMB -= delta.MemoryMB
+	r.DiskMB -= delta.DiskMB
+	r.IOPS -= delta.IOPS
+
+	for _, n := range delta.Networks {
+		// Find the matching interface by IP or CIDR
+		idx := r.NetIndex(n)
+		if idx == -1 {
+			r.Networks = append(r.Networks, n.Copy())
```

Comment:
Can you put a comment on this? I don't understand it as is, if it doesn't exist, we add it? Seems like the opposite of subtract!

Reply:
This was a copy paste bug, it should only subtract if the resource exists for the way I am using this.